### DISCUSSION TITLE
DB_PASSWORD_FILE and KEYCLOAK_PASSWORD_FILE on Quarkus

### DISCUSSION THREAD

**Discussion by madrover** [keycloak/keycloak#10938] [Archive]
Hi
Previously, on the Wildlfy distribution, we could [provide the some secrets via a file ](https://github.com/keycloak/keycloak-containers/tree/main/server#providing-the-username-and-password-via-files) using the DB_PASSWORD_FILE and KEYCLOAK_PASSWORD_FILE. That was really useful to use Docker Swarm secrets, for example.

I can see there's some support to file based secrets, as described at [Using Kubernetes secrets](https://www.keycloak.org/server/vault), but that seems to be a different use case.

What would be the Quarkus alternative for that?

I could do something like on our docker-compose file:
```
    command: |
      bash -c '
        export KEYCLOAK_ADMIN_PASSWORD=`cat /run/secrets/keycloak_password`
        /opt/keycloak/bin/kc.sh start --auto-build'
```
But it would be great something less hacky, as we had before.

Thanks!

**Comment by mdillenk** [comment]
Since the new way of obtaining the latest docker image of Keycloak is described here https://www.keycloak.org/server/containers

at the bottom instead of starting keycloak directly, like they do in the example
ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]

Do this instead

RUN mkdir /opt/keycloak/tools
COPY ["./docker-entrypoint-init", "/opt/keycloak/tools"]
ENTRYPOINT ["/opt/keycloak/tools/docker-entrypoint.sh"]  

that way a custom  entrypoint script will be part of your distro and in that entrtypoint file you can do something like this...

if [[ -z "${USING_SECRETS}" ]]; then
  echo "not using secrets for password"
else
  echo "using secrets set, will add password environment variable to system"
  export KC_DB_PASSWORD=`cat /run/secrets/db_password`
fi
echo "Starting Keycloak with SYS_PROPS: $SYS_PROPS"

exec /opt/keycloak/bin/kc.sh $KC_MODE $SYS_PROPS $@
exit $?
